File: MCRRTImp/src/main/java/lab/mars/MCRRTImp/PolarBasedImp/Bound.java
package lab.mars.MCRRTImp.PolarBasedImp;

import lab.mars.RRTBase.Obstacle;

public class Bound implements Obstacle<Polar> {

    double height;

    double width;

    public Bound(double height, double width) {
        this.height = height;
        this.width = width;
    }

    @Override
    public boolean contains(Polar o) {
        if (o.theta() > 90) {
            return false;
        }
        double x = o.r() * Math.cos(Math.toRadians(o.theta()));
        double y = o.r() * Math.sin(Math.toRadians(o.theta()));
        return x <= width && y <= height;
    }
}

File: MCRRTImp/src/test/kotlin/infrastructure/PolarBasedImp/Bound.kt
package infrastructure.PolarBasedImp

import lab.mars.MCRRTImp.base.Obstacle

class Bound(internal var height: Double, internal var width: Double) : Obstacle<Polar> {

    override operator fun contains(o: Polar): Boolean {
        if (o.theta() > 90) {
            return false
        }
        val x = o.r() * Math.cos(Math.toRadians(o.theta()))
        val y = o.r() * Math.sin(Math.toRadians(o.theta()))
        return x <= width && y <= height
    }
}
